src/disc/Albany_DiscretizationFactory.cpp

```diff
@@ -118,10 +118,7 @@ DiscretizationFactory::createMeshStruct(Teuchos::RCP<Teuchos::ParameterList> dis
           basal_params->set("Workset Size", -1);
         } else if (!basal_params->isParameter("Workset Size")) {
           // Compute basal workset size based on extruded workset size
-          const int num_cells_per_column_per_layer = disc_params->get<std::string>("Element Shape") ==
-              "Tetrahedron" ? 3 : 1; // number of 3D cells in one extruded 2D cell
-          const int num_cells_per_column = num_cells_per_column_per_layer * disc_params->get<int>("NumLayers");
-          int basal_ws_size = extruded_ws_size / num_cells_per_column;
+          int basal_ws_size = extruded_ws_size / disc_params->get<int>("NumLayers");
```

Comment:
This might create issues with the MPAS mesh struct. Need to check this branch with MPAS before you merge.